### PR TITLE
🚧 Add notification settings page

### PR DESCRIPTION
**Pull request by nwingt**
image

### CODE REVIEW

File: src/pages/settings/notification.vue [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+            </template>
+          </SettingsListItem>
+        </li>
+        <li>
+          <SettingsListItem :is-clickable="false">
+            {{ $t('settings_notification_tx_purchase') }}
+            <template #accessory>
+              <CheckBox v-model="isPurchaseNotificationEnabled"/>
+            </template>
+          </SettingsListItem>
```
Comment (williamchong) [inline src/pages/settings/notification.vue:21]: Can we set a threshold for purchase event to only notify if sales price is larger than a certain amount?

File: src/pages/settings/index.vue [hunk 2]
```diff
@@ -16,6 +16,11 @@
           <span class="settings-menu__item-title"> {{ $t('settings_following') }}</span>
         </NuxtLink>
       </li>
+      <li>
+        <NuxtLink class="settings-menu__item" :to="{ name: 'settings-notification' }">
+          <span class="settings-menu__item-title"> {{ $t('settings_notification') }}</span>
+        </NuxtLink>
+      </li>
```
Comment (williamchong) [inline src/pages/settings/index.vue:23]: Can merge with `email` page, setting email and notification seems highly related

File: src/pages/settings/notification.vue [hunk 3]
```diff
@@ -12,27 +12,73 @@
...
+  computed: {
+    ...mapGetters(['loginAddress', 'walletHasLoggedIn']),
+  },
+  watch: {
+    isTransferNotificationEnabled() {
+      if (this.isReady) {
+        this.handleToggle();
+      }
+    },
+  },
```
Comment (williamchong) [inline src/pages/settings/notification.vue:50]: Can we use `@change` on `CheckBox` instead of watch?

File: src/pages/settings/notification.vue [hunk 4]
```diff
@@ -12,27 +12,73 @@
...
+      try {
+        await this.signLogin();
+      } catch {
+        // No-op
+      }
+      if (!this.walletHasLoggedIn) {
+        return;
+      }
+    }
+    await this.fetchNotification();
```
Comment (williamchong) [inline src/pages/settings/notification.vue:62]: very confusing name, at least `fetchNotificationSettings` ?

File: src/pages/settings/notification.vue [hunk 5]
```diff
@@ -12,27 +12,73 @@
...
+    this.isReady = true;
+  },
+  methods: {
+    ...mapActions(['signLogin']),
+    async fetchNotification() {
+      const { data } = await this.$api.get(
+        getUserNotification(this.loginAddress)
```
Comment (williamchong) [inline src/pages/settings/notification.vue:69]: similarly `getUserNotificationSettings`

File: src/server/api/routes/users/v2/notification.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    try {
+      const { wallet: user } = req.params;
+      const userDoc = await walletUserCollection.doc(user).get();
+      const { notification = {} } = userDoc.data();
+      const { transfer = false } = notification;
+      res.json({
+        notification: {
+          transfer,
+        },
+      });
```
Comment (williamchong) [inline src/server/api/routes/users/v2/notification.js:27]: Just return `notification` object? preferably with a filter to filter out secret (if we do have)

File: src/pages/settings/notification.vue [hunk 7]
```diff
@@ -12,27 +12,73 @@
...
+      this.isTransferNotificationEnabled = transfer;
+    },
+    handleToggle() {
+      this.$api.post(getUserNotification(this.loginAddress), {
+        notification: {
+          transfer: this.isTransferNotificationEnabled,
+        },
+      });
+    },
+  },
```
Comment (williamchong) [inline src/pages/settings/notification.vue:82]: prefer generic function for this whole page, maybe `transfer` and `this.isTransferNotificationEnabled` as a input param when defining @change handler to call this function

**Review by williamchong — CHANGES_REQUESTED** [review]

File: src/pages/settings/notification.vue (hunk 5 above)
Comment (williamchong) [reply src/pages/settings/notification.vue:69]: Also consider putting this into `wallet` store

File: src/server/api/routes/users/v2/notification.js (hunk 6 above)
Comment (WeiJun0827) [reply src/server/api/routes/users/v2/notification.js:27]: Do we need to gerantee the fields in `notification` for this API? Or just let front-end decide the action if it's `undefined`?

Comment (williamchong) [reply src/server/api/routes/users/v2/notification.js:27]: yeah having defaults here could be useful, lets keep it then

File: src/pages/settings/notification.vue (hunk 7 above)
Comment (WeiJun0827) [reply src/pages/settings/notification.vue]: Seems `@input` event doesn't accept params, temporarily keep it.

**Pull request merged**